tests/components/wemo/test_fan.py

```diff
@@ -130,3 +135,47 @@ async def test_fan_set_humidity_service(
         blocking=True,
     )
     pywemo_device.set_humidity.assert_called_with(expected)
+
+
+@pytest.mark.parametrize(
+    "percentage,expected_fan_mode",
+    [
+        (0, FanMode.Off),
+        (10, FanMode.Minimum),
+        (30, FanMode.Low),
+        (50, FanMode.Medium),
+        (70, FanMode.High),
+        (100, FanMode.Maximum),
+    ],
+)
+async def test_fan_set_percentage(
+    hass, pywemo_device, wemo_entity, percentage, expected_fan_mode
+):
+    """Verify set_percentage works properly through the entire range of FanModes."""
+    assert await hass.services.async_call(
+        FAN_DOMAIN,
+        SERVICE_SET_PERCENTAGE,
+        {ATTR_ENTITY_ID: [wemo_entity.entity_id], ATTR_PERCENTAGE: percentage},
+        blocking=True,
+    )
+    pywemo_device.set_state.assert_called_with(expected_fan_mode)
+
+
+class TestInitialFanMode:
+    """Test that the FanMode is set to High when turned on the first time."""
+
+    @pytest.fixture
+    def pywemo_device(self, pywemo_device):
+        """Set the FanMode to off initially."""
+        pywemo_device.fan_mode = FanMode.Off
```

Comment:
Why not just set the attribute in the test?

Reply:
That would have been a good idea. It's convenient to use the `wemo_entity` entity fixture, which gets setup before the test method runs. Doing it this way avoids needing to add the entity in the test.

If you have ideas on improving the fixtures I'm happy to take the time an implement them. I appreciate that the tests are easier to follow if they follow the same style as the rest.